DXMainClient/Online/FileHashCalculator.cs

```diff
@@ -1,22 +1,26 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
 using ClientCore;
 using ClientCore.I18N;
-using DTAClient.Domain.Multiplayer;
+
 using Rampastring.Tools;
-using Utilities = Rampastring.Tools.Utilities;
 
 namespace DTAClient.Online
 {
     public class FileHashCalculator
     {
-        private FileHashes fh;
         private const string CONFIGNAME = "FHCConfig.ini";
         private bool calculateGameExeHash = true;
 
-        string[] fileNamesToCheck = new string[]
+        private static readonly IReadOnlyList<string> knownTextFileExtensions = [".txt", ".ini", ".json", ".xml"];
```

Comment:
I am not sure if this is a robust impl, what do we risking if a used text file is not included here?

Reply:
This design is mostly for preventing an honest-but-curious user viewing a text file bundled with a mod, and their text editor (probably with an auto saving, or one could accidentally hit Ctrl+S even if they think they haven't change any single character) might save that file with the line breaks normalized (they might be non-consistent due to collaboration in Git).

If a text file does not have the known extensions here, the behavior should be similar with the one as if this PR is not merged, i.e., be treated as a binary file.